edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs

```diff
@@ -125,6 +126,32 @@ async Task BeginUploadRetry(CancellationToken cancellationToken)
             }
         }
 
+        /// <summary>
+        /// Replaces the device id in some edgeHub metrics with "device".
+        ///
+        /// EdgeHub metrics id comes in the form of 'deviceId/moduleName' in the case of an edgeDevice
+        /// and 'deviceId' in the case of downstream leaf devices.
+        /// </summary>
+        /// <param name="id">Metric id tag.</param>
+        /// <returns>Id tag with deviceId removed.</returns>
+        string ReplaceDeviceId(string id)
+        {
+            const string deviceIdReplacement = "device";
+
+            // Id is in the form of 'deviceId/moduleId'
+            string[] parts = id.Split('/');
+            if (parts.Length == 2)
+            {
+                parts[0] = deviceIdReplacement;
+                parts[1] = parts[1].CreateSha256(); // Hash moduleId
```

Comment:
Any benefit to **NOT** hashing the module ID if it's a system module (i.e., begins with `$`)?

Reply:
easier to read, I'll change it.